lib/resty/iprepd.lua

```diff
@@ -191,17 +199,39 @@ end
 
 function _M.audit_log(self, ip)
   if self.audit_blocked_requests == 1 then
-    ngx.req.read_body()
+    local should_audit_request = 0
     local headers = ngx.req.get_headers()
-    local request_headers_all = ""
-    for k,v in pairs(headers) do
-      request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+    for i, uri in ipairs(self.audit_uri_list) do
+      if string.match(ngx.var.uri, uri) then
+        should_audit_request = 1
+        break
+      end
     end
-    local data = ngx.req.get_body_data()
-    if not data then
-      data = ngx.req.get_body_file()
+    if should_audit_request == 1 then
+      ngx.req.read_body()
+      local request_headers_all = ""
+      if self.audit_include_headers == 1 then
+        local headers = ngx.req.get_headers()
+        for k,v in pairs(headers) do
+          if k == "authorization" or k == "proxy-authorization" then
```

Comment:
Performance wise it would likely be much faster to just nil the value out of the table rather then do this check for every header, but since this option is for debugging this should be fine.

Reply:
made the change.